nni/retiarii/nn/pytorch/api.py

```diff
@@ -64,9 +65,18 @@ def create_fixed_module(cls, candidates: Union[Dict[str, nn.Module], List[nn.Mod
                             label: Optional[str] = None, **kwargs):
         chosen = get_fixed_value(label)
         if isinstance(candidates, list):
-            return candidates[int(chosen)]
+            result = candidates[int(chosen)]
         else:
-            return candidates[chosen]
+            result = candidates[chosen]
+
+        # map the named hierarchies to support weight inheritance for python engine
+        if hasattr(result, STATE_DICT_PY_MAPPING_PARTIAL):
+            # already has a mapping, will merge with it
+            prev_mapping = getattr(result, STATE_DICT_PY_MAPPING_PARTIAL)
+            setattr(result, STATE_DICT_PY_MAPPING_PARTIAL, {k: f'{chosen}.{v}' for k, v in prev_mapping.items()})
+        else:
+            setattr(result, STATE_DICT_PY_MAPPING_PARTIAL, {'': str(chosen)})
+        return result
```

Comment:
I failed to connect this part with supernet and state dict load/dump, maybe you can brief this pr in a quick discussion

Reply:
The modules has an attribute called `STATE_DICT_PY_MAPPING_PARTIAL`, which records which variables belongs to which module originally. Then the hooks use those attributes to recover the full mapping, so as to map the original state dict into the current module hierarchy.